### PR TITLE
Updated menu swapper to include a make-all feature

### PR DESCRIPTION
**Pull request by Pashweetie**
Added make all option in response to 
https://github.com/runelite/runelite/issues/3741

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/RuneLite.java [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
+//* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+//* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+//* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+//* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+//* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+//* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+//* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+//* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (bmiddle) [inline runelite-client/src/main/java/net/runelite/client/RuneLite.java]: Block comment was fine, there's no need to make the copyright header a bunch of single-line comments

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 2]
```diff
@@ -232,4 +232,11 @@ default boolean swapTravel()
 	{
 		return true;
 	}
+	@ConfigItem(
+			position=18,
```
Comment (bmiddle) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: Need whitespace around the equals

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 3]
```diff
@@ -232,4 +232,11 @@ default boolean swapTravel()
 	{
 		return true;
 	}
+	@ConfigItem(
+			position=18,
+			keyName="makeAll",
+			name = "Make All",
+			description ="Some things the new make all doesn't work for, like jewelry, this fixes most if not all of those"
+	)
+	default boolean makeAll(){return true;}
```
Comment (bmiddle) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: Braces need to be on their own lines

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 4]
```diff
@@ -328,6 +328,10 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		int itemId = event.getIdentifier();
 		String option = Text.removeTags(event.getOption()).toLowerCase();
 		String target = Text.removeTags(event.getTarget()).toLowerCase();
+		if (config.makeAll()&& option.equals("make-1"))
+		{
+			swap("make-all",option,target,true);
```
Comment (bmiddle) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Spaces after a comma

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 5]
```diff
@@ -328,6 +328,10 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		int itemId = event.getIdentifier();
 		String option = Text.removeTags(event.getOption()).toLowerCase();
 		String target = Text.removeTags(event.getTarget()).toLowerCase();
+		if (config.makeAll()&& option.equals("make-1"))
```
Comment (bmiddle) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Space after `makeAll()`

File: runelite-client/src/main/java/net/runelite/client/RuneLite.java (hunk 1 above)
Comment (Pashweetie) [reply runelite-client/src/main/java/net/runelite/client/RuneLite.java]: Yeah i just had an outdated base

Comment (Pashweetie) [reply runelite-client/src/main/java/net/runelite/client/RuneLite.java]: How do i update the PR with my new code

Comment (bmiddle) [reply runelite-client/src/main/java/net/runelite/client/RuneLite.java]: If you push your code into the same branch it'll automatically show up

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 6]
```diff
@@ -328,6 +328,10 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		int itemId = event.getIdentifier();
 		String option = Text.removeTags(event.getOption()).toLowerCase();
 		String target = Text.removeTags(event.getTarget()).toLowerCase();
+		if (config.makeAll() && option.equals("make-1"))
```
Comment (MagicfTail) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Add empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 7]
```diff
@@ -232,4 +232,15 @@ default boolean swapTravel()
 	{
 		return true;
 	}
+
+
```
Comment (MagicfTail) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: remove empty line

File: runelite-client/src/main/java/net/runelite/client/RuneLite.java [hunk 8]
```diff
@@ -1,27 +1,29 @@
...
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
```
Comment (MagicfTail) [inline runelite-client/src/main/java/net/runelite/client/RuneLite.java]: remove empty line

File: runelite-client/src/main/java/net/runelite/client/RuneLite.java [hunk 9]
```diff
@@ -1,27 +1,29 @@
...
```
Comment (MagicfTail) [inline runelite-client/src/main/java/net/runelite/client/RuneLite.java]: Remove empty line

Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/RuneLite.java]: Undo all of your changes to this file; no code changes were made here.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java (hunk 7 above)
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]: Remove the extra newline here--there should be only one empty line between config items.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java [hunk 10]
```diff
@@ -232,4 +232,15 @@ default boolean swapTravel()
...
+
+	@ConfigItem(
+			position = 18,
+			keyName = "makeAll",
+			name = "Make all",
+			description = "Some things the new make all doesn't work for, like jewelry, this fixes most if not all of those"
+	)
+	default boolean makeAll()
+	{ return true; }
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperConfig.java]:
```
{
    return true;
}
```

**Comment by graingert** [comment]
@Pashweetie any chance of rebasing? There's been upstream changes that have caused conflicts

**Comment by chrisjohnson** [comment]
@Pashweetie would love to see this (or something like it) merged in, any chance you could revisit it?

**Comment by deathbeam** [comment]
I believe this feature was rejected:

```
Menu reorder in shop/artisan skill screens: Other clients don't implement these swaps, and would be too strong for numerous activities (examples include "Buy-50" swaps, "Make All" swaps for smithing/crafting, etc.)
```

**Pull request closed without merging**